packages/lib/file-api-driver-amazon-s3.js

```diff
@@ -3,7 +3,7 @@ const { basename } = require('./path-utils');
 const shim = require('./shim').default;
 const JoplinError = require('./JoplinError').default;
 const { Buffer } = require('buffer');
-const { Readable } = require('stream').Readable;
+//const Response = require('node-fetch').Response;
```

Comment:
we are at a sort of impasse here that I don't know how best to resolve. 

Desktop needs this line, where react-native on iOS doesn't.

@roman-r-m if you want to pull and try both (uncomment for desktop / comment for android) to confirm the behaviour is the same. I'm not sure how to do a conditional import or if there is some other way to load this only on desktop.

Reply:
I'll give it a try today after work.

Something like `shim.isNode/isElectron/isReactNative` might help here.

Reply:
Indeed usually we use the shim, and in particular shim-init-node and shim-init-react to load certain things on desktop and others on mobile. See for example `shim.Geolocation`

Reply:
Seems to work fine on Android, I can sync both ways.

Reply:
awesome! I'll take a look to see how I can shim that in appropriately for desktop.

Reply:
@laurent22 thanks for that great example with Geolocation. In my latest commits I've used the shim and it's working in my env on desktop and iOS. 🎉 

I'm going to run the tests, and @roman-r-m if you could pull latest and confirm it works for you on desktop/android that would be appreciated. ❤️